src/vfs2.c

```diff
@@ -907,7 +1028,7 @@ static int vfs2_shm_unmap(sqlite3_file *file, int delete)
 	(void)delete;
```

Comment:
This is also imitated from the old VFS. SQLite sets this flag when it's planning to delete the WAL from disk (because it has been completely checkpointed and the last connection is closing, see [here](https://github.com/sqlite/sqlite/blob/07f215ad9e3606253d9bc52d59e46a9c7d9a1b63/src/wal.c#L2517)), and it wants the WAL-index to be deleted as well. In dqlite we set [`SQLITE_DBCONFIG_NO_CKPT_ON_CLOSE`](https://sqlite.org/c3ref/c_dbconfig_defensive.html) and we implement the [PERSIST_WAL](https://sqlite.org/c3ref/c_fcntl_begin_atomic_write.html#sqlitefcntlpersistwal) fcntl so that SQLite never tries to delete the WAL, so we don't expect it to send us this flag.

Reply:
Having written that, I realize it's not clear why this shouldn't be `assert(delete == 0)`. I think I tried rewriting it that way at one point and it caused something to go wrong, but I don't remember what. I'll experiment and write a comment about this.

Reply:
You read my mind in the second comment :P